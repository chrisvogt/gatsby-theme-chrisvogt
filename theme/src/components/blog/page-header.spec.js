import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';
import PageHeader from './page-header';

describe('PageHeader', () => {
  it('renders correctly with given children', () => {
    render(<PageHeader>Hello, World!</PageHeader>);
    const headingElement = screen.getByRole('heading', { name: /Hello, World!/i });
    expect(headingElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(<PageHeader>Hello, World!</PageHeader>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
