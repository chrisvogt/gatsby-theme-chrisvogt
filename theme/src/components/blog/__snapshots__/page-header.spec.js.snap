// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader matches snapshot 1`] = `
<h1
  className="p-name css-1sus4jf"
>
  Hello, World!
</h1>
`;
